### ISSUE TITLE
JDBCPreparedQuery fails to handle parameters of type Buffer

### ISSUE THREAD

**Issue by riemenschneider** [vert-x3/vertx-jdbc-client#210]
### Version

4.0.2

### Context

I used the new SQL client API for creating a pool together with prepared statements. Unfortunately tis breaks as soon as I start using Buffer parameters.
When using the following code:
```
pool.preparedQuery("INSERT INTO test_table ( test_data ) VALUES ( ? )")
	.execute(Tuple.of(Buffer.buffer("Hello world!")))
	.onFailure(e -> log.error("Failed to insert buffer into table.", e))
```
an exception is thrown addressing an invalid position argument 0.
This is caused by https://github.com/vert-x3/vertx-jdbc-client/blob/cae45dafab9121ed527d23227c895415b977656f/src/main/java/io/vertx/jdbcclient/impl/actions/JDBCPreparedQuery.java#L154 which uses an illegal argument according the the called method's Java API.
> **Parameters:**
> **pos** the position in the BLOB object at which to start writing; the first position is 1

### Do you have a reproducer?

A reproducer can be found at https://github.com/riemenschneider/JDBCPreparedQueryTest.

### Steps to reproduce

1. Clone the reproducer.
2. Run `gradlew test` in the root folder of the cloned reproducer.

**Comment by vietj** [comment]
assigned to you @pmlopes

**Comment by tsegismont** [comment]
Fixed in #212
